src/test/java/org/opensearch/security/configuration/SettingsPermissionValveTests.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.configuration;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.opensaml.xmlsec.signature.P;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.security.DefaultObjectMapper;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.test.SingleClusterTest;
+import org.opensearch.security.test.helper.rest.RestHelper;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SettingsPermissionValveTests extends SingleClusterTest {
```

Comment:
I would recommend to use the int test style like it is used here:

https://github.com/opensearch-project/security/blob/main/src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java

That way, you can create all configuration using a declarative style. You won't need to create a new directory containing test resources then (experience has shown that these test resource directories are quite hard to manage).